pkg/sql/catalog/seqexpr/sequence.go

```diff
@@ -80,23 +80,21 @@ func GetSequenceFromFunc(funcExpr *tree.FuncExpr) (*SeqIdentifier, error) {
 
 	if hasSequenceArguments {
 		found := false
-		for i := range def.Overloads {
+		for _, overload := range def.Overloads {
 			// Find the overload that matches funcExpr.
-			if len(funcExpr.Exprs) == def.Overloads[i].Types.Length() {
-				found = true
-				paramTypes, ok := def.Overloads[i].Types.(tree.ParamTypes)
+			if len(funcExpr.Exprs) == overload.Types.Length() {
+				paramTypes, ok := overload.Types.(tree.ParamTypes)
 				if !ok {
 					panic(pgerror.Newf(
 						pgcode.InvalidFunctionDefinition,
 						"%s has invalid argument types", funcExpr.Func.String(),
 					))
 				}
-				for i := 0; i < def.Overloads[i].Types.Length(); i++ {
+				found = true
+				for i := 0; i < len(paramTypes); i++ {
 					// Find the sequence name param.
-					paramName := paramTypes[i].Name
-					if paramName == builtinconstants.SequenceNameArg {
-						arg := funcExpr.Exprs[i]
-						if seqIdentifier := getSequenceIdentifier(arg); seqIdentifier != nil {
+					if paramTypes[i].Name == builtinconstants.SequenceNameArg {
+						if seqIdentifier := getSequenceIdentifier(funcExpr.Exprs[j]); seqIdentifier != nil {
```

Comment:
nit: it might be easier to unserstand the code if you keep the `arg := funcExpr.Exprs[i]` declaration

Reply:
good point, done.